packages/client/src/features/smart-connector/smart-connector.ts

```diff
@@ -0,0 +1,524 @@
+/********************************************************************************
+ * Copyright (c) 2023 Business Informatics Group (TU Wien) and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { inject, injectable } from 'inversify';
+import { GLSPActionDispatcher } from '../../base/action-dispatcher';
+import { EditorContextService } from '../../base/editor-context-service';
+import { FocusTracker } from '../../base/focus/focus-tracker';
+import {
+    AbstractUIExtension,
+    Action,
+    IActionHandler,
+    ICommand,
+    KeyListener,
+    GModelElement,
+    GModelRoot,
+    matchesKeystroke,
+    OpenSmartConnectorAction,
+    SetUIExtensionVisibilityAction,
+    ViewportResult,
+    RequestContextActions,
+    SetContextActions,
+    PaletteItem,
+    TriggerEdgeCreationAction,
+    TriggerNodeCreationAction,
+    Args,
+    SmartConnectorGroupItem,
+    SmartConnectorNodeItem,
+    SmartConnectorPosition,
+    KeyCode
+} from '@eclipse-glsp/sprotty';
+import { GetViewportAction } from 'sprotty-protocol/lib/actions';
+import { changeCodiconClass, createIcon } from '../tool-palette/tool-palette';
+import { IDiagramStartup } from '../../base/model/diagram-loader';
+import { EnableDefaultToolsAction } from '../../base/tool-manager/tool';
+
+const CONTAINER_PADDING_PX = 20;
+const MAX_HEIGHT_GROUP = '150px';
+const SEARCH_FIELD_SUFFIX = '_search_field';
+const SMART_CONNECTOR_CONTAINER_ID = 'smart-connector-container';
+const EXPAND_BUTTON_ID = 'smart-connector-expand-button';
+const GROUP_CONTAINER_CLASS = 'smart-connector-group-container';
+const HEADER_CLASS = 'smart-connector-group-header';
+const GROUP_CLASS = 'smart-connector-group';
+const TOOL_BUTTON_CLASS = 'smart-connector-button';
+const COLLAPSABLE_CLASS = 'collapsable-group';
```

Comment:
```suggestion
export namespace SmartConnector {
export const CONTAINER_PADDING_PX = 20;
export const MAX_HEIGHT_GROUP = '150px';
export const SEARCH_FIELD_SUFFIX = '_search_field';
export const SMART_CONNECTOR_CONTAINER_ID = 'smart-connector-container';
export const EXPAND_BUTTON_ID = 'smart-connector-expand-button';
export const GROUP_CONTAINER_CLASS = 'smart-connector-group-container';
export const HEADER_CLASS = 'smart-connector-group-header';
export const GROUP_CLASS = 'smart-connector-group';
export const TOOL_BUTTON_CLASS = 'smart-connector-button';
export const COLLAPSABLE_CLASS = 'collapsable-group';
}
```

If we export it as suggested here, then others can also reuse the values.